golang/cosmos/x/swingset/keeper/snapshotter.go

```diff
@@ -0,0 +1,384 @@
+package keeper
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/Agoric/agoric-sdk/golang/cosmos/vm"
+	"github.com/Agoric/agoric-sdk/golang/cosmos/x/swingset/types"
+	vstoragetypes "github.com/Agoric/agoric-sdk/golang/cosmos/x/vstorage/types"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	snapshots "github.com/cosmos/cosmos-sdk/snapshots/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+var _ snapshots.ExtensionSnapshotter = &SwingsetSnapshotter{}
+
+// SnapshotFormat 1 is a proto message containing an artifact name, and the binary artifact data
+const SnapshotFormat = 1
+
+const ExportManifestFilename = "export-manifest.json"
+const ExportDataFilename = "export-data.json"
+const UntrustedExportDataArtifactName = "@@@### Untrusted Export Data ###@@@"
```

Comment:
I'd prefer something just as obvious but less magical sounding, like:
```suggestion
const UntrustedExportDataArtifactName = "UNTRUSTED-EXPORT-DATA"
```

Reply:
I wanted something that would obviously be an uncommon filename, but I've added sanitation of artifact names since, so I agree, better use something less magical.